amr-wind/equation_systems/DiffusionOps.cpp

```diff
@@ -57,30 +57,40 @@ void DiffSolverIface<LinOp>::setup_operator(
     BL_PROFILE("amr-wind::setup_operator");
     auto& repo = m_pdefields.repo;
     const int nlevels = repo.num_active_levels();
-    auto& density = m_density.state(fstate);
 
     linop.setScalars(alpha, beta);
     for (int lev = 0; lev < nlevels; ++lev) {
-        linop.setACoeffs(lev, density(lev));
         linop.setLevelBC(lev, &m_pdefields.field(lev));
     }
+    this->set_acoeffs(linop, fstate);
     set_bcoeffs(linop);
 }
 
-template <typename LinOp>
+template<typename LinOp>
+void DiffSolverIface<LinOp>::set_acoeffs(LinOp& linop, const FieldState fstate)
+{
+    BL_PROFILE("amr-wind::set_acoeffs");
+    auto& repo = m_pdefields.repo;
+    const int nlevels = repo.num_active_levels();
+    auto& density = m_density.state(fstate);
+
+    for (int lev = 0; lev < nlevels; ++lev) {
+        linop.setACoeffs(lev, density(lev));
+    }
+}
+
+template<typename LinOp>
 void DiffSolverIface<LinOp>::setup_solver(amrex::MLMG& mlmg)
 {
     BL_PROFILE("amr-wind::setup_solver");
     // Set all MLMG options
     m_options(mlmg);
 }
 
-template <typename LinOp>
-void DiffSolverIface<LinOp>::linsys_solve(const amrex::Real dt)
+template<typename LinOp>
+void DiffSolverIface<LinOp>::linsys_solve_impl()
```

Comment:
Need to make the visibility of this method `public` to allow building on CUDA.

Reply:
Huh? https://github.com/Exawind/amr-wind/blob/main/amr-wind/equation_systems/DiffusionOps.cpp#L79